src/Generating/StructGen.cpp

```diff
@@ -34,16 +41,12 @@ void cStructGenTrees::GenFinish(cChunkDesc & a_ChunkDesc)
 
 			if ((x != 1) || (z != 1))
 			{
-				Dest = &WorkerDesc;
-				WorkerDesc.SetChunkCoords({BaseX, BaseZ});
-
-				// TODO: This may cause a lot of wasted calculations, instead of pulling data out of a single (cChunkDesc) cache
-
-				cChunkDesc::Shape workerShape;
-				m_BiomeGen->GenBiomes           ({BaseX, BaseZ}, WorkerDesc.GetBiomeMap());
-				m_ShapeGen->GenShape            ({BaseX, BaseZ}, workerShape);
-				WorkerDesc.SetHeightFromShape   (workerShape);
-				m_CompositionGen->ComposeTerrain(WorkerDesc, workerShape);
+				Dest = WorkerDesc.get();
+				WorkerDesc->SetChunkCoords({BaseX, BaseZ});
+				m_BiomeGen->GenBiomes           ({BaseX, BaseZ}, WorkerDesc->GetBiomeMap());
+				m_ShapeGen->GenShape            ({BaseX, BaseZ}, WorkerShape->mShape);
+				WorkerDesc->SetHeightFromShape   (WorkerShape->mShape);
```

Comment:
Is the alignment here intentional, it looks like there's an extra space

Reply:
I think it's a typo, but the alignment here is ugly and against conventional style so should probably just be removed entirely.